src/scene_battle_rpg2k.cpp

```diff
@@ -467,52 +517,31 @@ bool Scene_Battle_Rpg2k::ProcessBattleAction(Game_BattleAlgorithm::AlgorithmBase
 				Game_System::SePlay(*action->GetStartSe());
 			}
 
-			battle_action_state = BattleActionState_Result;
+			action->Apply();
+			battle_action_state = BattleActionState_ResultPop;
 
-			break;
-		case BattleActionState_ConditionHeal:
-			if (action->IsFirstAttack()) {
-				std::vector<int16_t> states_to_heal = action->GetSource()->NextBattleTurn();
-				std::vector<int16_t> states_remaining = action->GetSource()->GetInflictedStates();
-				action->GetSource()->ApplyConditions();
-				bool message_to_show = false;
-				if (!states_to_heal.empty() || !states_remaining.empty()) {
-					battle_message_window->Clear();
-					for (auto state_id : states_to_heal) {
-						// BattleAlgorithm verifies the states
-						const RPG::State* state = ReaderUtil::GetElement(Data::states, state_id);
-						if (!state->message_recovery.empty()) {
-							battle_message_window->PushWithSubject(state->message_recovery, action->GetSource()->GetName());
-							message_to_show = true;
-						}
-					}
-					for (auto state_id : states_remaining) {
-						// BattleAlgorithm verifies the states
-						const RPG::State* state = ReaderUtil::GetElement(Data::states, state_id);
-						if (!state->message_affected.empty()) {
-							battle_message_window->PushWithSubject(state->message_affected, action->GetSource()->GetName());
-							message_to_show = true;
-						}
-					}
-					if (message_to_show) {
-						battle_action_wait = GetDelayForWindow();
-					}
-					else {
-						battle_action_wait = 0;
-					}
-				}
-				else {
-					battle_action_wait = 0;
-				}
+			if (!action->IsFirstAttack()) {
+				battle_action_wait = 0;
+				return ProcessBattleAction(action);
 			}
 
-			battle_action_state = BattleActionState_Start;
-
 			break;
-		case BattleActionState_Result:
-			battle_action_wait = GetDelayForWindow();
+		case BattleActionState_ResultPop:
+			battle_action_wait = std::min<int>(GetDelayForLine() / 2, battle_action_wait);
 
 			if (battle_result_messages_it != battle_result_messages.end()) {
+				default_result_lines = 1 + (action->HasSecondStartMessage() ? 1 : 0);
+				while (battle_message_window->GetLineCount() > (default_result_lines + *battle_result_order_it))
+					battle_message_window->Pop();
```

Comment:
All this ugly out_vector stuff just for this while-loop thing which will always add 0 except for 1 case... oh come on Sormat :/